base_layer/core/src/base_node/sync/block_sync/synchronizer.rs

```diff
@@ -267,4 +280,21 @@ impl<B: BlockchainBackend + 'static> BlockSynchronizer<B> {
         .await
         .expect("block validator panicked")
     }
+
+    async fn ban_peer<T: ToString>(&mut self, node_id: NodeId, reason: T) -> Result<(), BlockSyncError> {
+        let reason = reason.to_string();
+        if self.config.sync_peers.contains(&node_id) {
+            debug!(
+                target: LOG_TARGET,
+                "Not banning peer that is allowlisted for sync. Ban reason = {}", reason
+            );
+            return Ok(());
+        }
```

Comment:
Should this logic rather not live in the connectivity manager? Here we don't ban a peer because it is allow listed, but I don't think we have the same logic with propagation, there that peer will get banned.

Reply:
This is the force sync peer setting specific to sync, not the seed peers. By default it's empty.